libs/sdk-ui-ext/src/internal/components/configurationPanels/BarChartConfigurationPanel.tsx

```diff
@@ -1,9 +1,26 @@
-// (C) 2019 GoodData Corporation
+// (C) 2019-2022 GoodData Corporation
 import BaseChartConfigurationPanel from "./BaseChartConfigurationPanel";
 import { BAR_CHART_AXIS_CONFIG } from "../../constants/axis";
+import { IAxisProperties } from "../../interfaces/AxisType";
+import { countItemsOnAxes } from "../pluggableVisualizations/baseChart/insightIntrospection";
 
 export default class BarChartConfigurationPanel extends BaseChartConfigurationPanel {
     protected getAxesConfiguration(type: string): any[] {
         return BAR_CHART_AXIS_CONFIG[type];
     }
+
+    protected isNameSubsectionDisabled(axis: IAxisProperties) {
```

Comment:
this method is repeated three times in three different classes, use only single shared implementation. Eg by some parent class